controllers/storagecluster/reconcile.go

```diff
@@ -318,7 +318,7 @@ func (r *StorageClusterReconciler) reconcilePhases(
 			}
 		}
 
-		if err := r.reconcileUninstallAnnotations(instance); err != nil {
+		if isUpdated, err := r.reconcileUninstallAnnotations(instance); err != nil || isUpdated {
 			return reconcile.Result{}, err
 		}
```

Comment:
Let's have some kind of log message (maybe info) attached to this kind of early return.
Also `isUpdated` -> `hasUpdated` or even better scWasUpdated

Reply:
done